src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -0,0 +1,118 @@
+    permission_classes = (OrganizationPermission,)
+
+    def has_feature(self, request, organization):
+        return features.has("organizations:discover", organization, actor=request.user)
+
+    def get_project(self, organization, project_id):
+        """ Get a project for an org, or None if one doesn't exist """
+        return Project.objects.filter(
+            organization=organization, status=ProjectStatus.VISIBLE, id=project_id
+        ).first()
+
+    def post(self, request, organization):
+        """ Create A Key Transaction """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        project = self.get_project(organization, int(request.data["project"]))
+
+        if project is None:
+            return Response({"detail": "No project with that id found"}, status=400)
+
+        base_filter = {"organization": organization, "project": project, "owner": request.user}
+
+        # Limit the number of key transactions
+        if KeyTransaction.objects.filter(**base_filter).count() >= MAX_KEY_TRANSACTIONS:
+            return Response(
+                {"detail": "At most {} Key Transactions can be added".format(MAX_KEY_TRANSACTIONS)},
+                status=400,
+            )
+
+        base_filter["transaction"] = request.data["transaction"]
+        if KeyTransaction.objects.filter(**base_filter).count() == 1:
+            return Response({"detail": "This Key Transaction was already added"}, status=400)
+
+        KeyTransaction.objects.create(**base_filter)
+        return Response(status=201)
+
+    def get(self, request, organization):
+        """ Get the paged Key Transactions for a user """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        params = self.get_filter_params(request, organization)
+        fields = request.GET.getlist("field")[:]
+        orderby = self.get_orderby(request)
+
+        queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
+
+        results = query(
+            fields,
+            None,
+            params,
+            orderby=orderby,
+            referrer="discover.key_transactions",
+            # The snuba query for transactions is of the form
+            # (transaction="1" AND project=1) OR (transaction="2" and project=2) ...
+            # which the schema intentionally doesn't support so we cannot do an AND in OR
+            # so here the "and" operator is being instead to do an AND in OR query
+            conditions=[
+                [
+                    # First layer is Ands
+                    [
+                        # Second layer is Ors
+                        [
+                            "and",
+                            [
+                                [
+                                    "equals",
+                                    ["transaction", u"'{}'".format(transaction.transaction)],
```

Comment:
What happens if the transaction name contains quotes or ends in a backslash?

Reply:
`equals` needs quotes around this value to think that the value there is a string and not another column.

And 🤔 huh now you bring it up I thought it wouldn't work too, but I think `json.dumps` prevents anything bad from happening.
eg. 
```
json.dumps(["transaction", "'{}'".format('bad"\\')])
'["transaction", "\'bad\\"\\\\\'"]'
```

Added some tests to check this too 👍

Reply:
-- nevermind looks like I just wrote a flakey test 😞 seems to only pass intermittently.

Reply:
Per our other conversation, if snuba is not parameterizing these values then this is a sql injection risk.

Reply:
Here's what I tested to check that this is fine using the snuba query interface. The value is set inside an equals like this:
```
[
    "equals",
        ["transaction", INPUT]
    ],
]
```

|INPUT|OUTPUT|notes|
|------|---------|------|
|`some'thing`|`WHERE (((transaction = 'some\\'thing')`|Mid word quote|
|`'; DROP TABLE sentry_local;`|`WHERE (((transaction = '\\'; DROP TABLE sentry_local;')`|classic drop table|
|`something\`|Errors with `"Invalid \\X escape sequence '\\\\': line 50 column 28 (char 909)"`|Forcing a backslash in, this shouldn't happen anyways since json.dumps will escape the slash|
|`something\\`|`WHERE (((transaction = 'something\\\\')`|What a transaction with a trailing `\` would actually look like|
|`some’thing`|`WHERE (((transaction = 'some’thing')`|Homoglyph|
|`some”thing`|`WHERE (((transaction = 'some”thing')`|Another Homoglyph|

Reply:
Thanks for looking into this :clap: